### PR TITLE
templates: Add `fileStat` function

### PR DESCRIPTION
**Pull request by taophp**
May be useful in a loop handling listFiles entries to adopt different behavior is a file is a directory.

### CODE REVIEW

File: modules/caddyhttp/templates/tplcontext.go [hunk 1]
```diff
@@ -79,6 +79,7 @@ func (c *TemplateContext) NewTemplate(tplName string) *template.Template {
 		"markdown":         c.funcMarkdown,
 		"splitFrontMatter": c.funcSplitFrontMatter,
 		"listFiles":        c.funcListFiles,
+		"isDir":            c.isDir,
```
Comment (mohammed90) [inline modules/caddyhttp/templates/tplcontext.go:82]:
```suggestion
		"isDir":            c.funcIsDir,
```

**Comment by francislavoie** [comment]
The CLA isn't happy because the email on your git commits doesn't match the ones on your Github account. Make sure to fix that!

**Comment by taophp** [comment]
> The CLA isn't happy because the email on your git commits doesn't match the ones on your Github account. Make sure to fix that!

Fixed ! Thanks and sorry.

**Comment by mholt** [comment]
Should we just return the file info instead of only the IsDir result? Then the user can get file size, etc.

**Comment by taophp** [comment]
> Thanks for the contribution!

My pleasure @mholt .

> But maybe it would make more sense to return the whole file info instead of only the IsDir result? Then the user can get file size, etc.

Well, this surely is a good idea. But, in my use case, I do not need it currently. And, as it's the first time I code in Go, I do not feel very confident to handle this job : my contribution is essentially a rewrite of the `funcFileExists` function... It's up to you to add a more complete `fileinfo` function.

If you decide to write it, I should still appreciate that you keep the `isDir` function: apart from the honor for me to become a contributor to Caddy, I find it really practical to use, and I would doubtless not be the only one; moreover, it seems consistent with the Go language.

File: modules/caddyhttp/templates/tplcontext.go [hunk 2]
```diff
@@ -79,6 +79,7 @@ func (c *TemplateContext) NewTemplate(tplName string) *template.Template {
 		"markdown":         c.funcMarkdown,
 		"splitFrontMatter": c.funcSplitFrontMatter,
 		"listFiles":        c.funcListFiles,
+		"isDir":            c.funcIsDir,
```
Comment (mholt) [inline modules/caddyhttp/templates/tplcontext.go:82]:
```suggestion
		"stat":            c.funcIsDir,
```

(Maybe rename `funcIsDir` to `funcStat`)

**Comment by taophp** [comment]
@mholt OK, thanks for you help. I committed an new version, improved in the way you suggested.

**Comment by taophp** [comment]
Sorr @francislavoie , I just forgot to remove my first function...

**Comment by taophp** [comment]
> Have you tried it yet to make sure it does what you need? (What does your relavent template code look like?)

Yes, I did, and it works fine ! Thanks for your help. My code in the template looks like:
```
{{- $listFile := listFiles "/" -}}
{{- range $filePath := $listFile -}}
	{{ $stat := stat $filePath }}
	{{ $stat.IsDir }}
{{- end -}}
```

**Pull request merged**